gtk/tests_backend/app.py

```diff
@@ -25,25 +25,42 @@ def __init__(self, app):
         assert IS_WAYLAND is (os.environ.get("WAYLAND_DISPLAY", "") != "")
 
     @property
-    def config_path(self):
-        return Path.home() / ".config/testbed"
-
-    @property
-    def data_path(self):
-        return Path.home() / ".local/share/testbed"
-
-    @property
-    def cache_path(self):
-        return Path.home() / ".cache/testbed"
-
-    @property
-    def logs_path(self):
-        return Path.home() / ".local/state/testbed/log"
+    def _custom_path_root(self):
+        return Path.home() / ".cache/testbed/custom_app_paths"
+
+    def paths(self, *, custom):
+        if custom:
+            return {
+                "config": self._custom_path_root / "config/testbed",
+                "data": self._custom_path_root / "data/testbed",
+                "cache": self._custom_path_root / "cache/testbed",
+                "logs": self._custom_path_root / "state/testbed/log",
+            }
+
+        return {
+            "config": Path.home() / ".config/testbed",
+            "data": Path.home() / ".local/share/testbed",
+            "cache": Path.home() / ".cache/testbed",
+            "logs": Path.home() / ".local/state/testbed/log",
+        }
 
     @property
     def is_cursor_visible(self):
         pytest.skip("Cursor visibility not implemented on GTK")
 
+    def apply_path_customization(self):
+        root = Path.home() / ".cache/testbed/custom_app_paths"
+        os.environ["XDG_CONFIG_HOME"] = str(root / "config")
+        os.environ["XDG_DATA_HOME"] = str(root / "data")
+        os.environ["XDG_CACHE_HOME"] = str(root / "cache")
+        os.environ["XDG_STATE_HOME"] = str(root / "state")
+
+    def remove_path_customization(self):
+        del os.environ["XDG_CONFIG_HOME"]
+        del os.environ["XDG_DATA_HOME"]
+        del os.environ["XDG_CACHE_HOME"]
+        del os.environ["XDG_STATE_HOME"]
```

Comment:
This might be better as a `contextlib.contextmanager`, so it can't be accidentally mis-used.

Reply:
Done.